src/Raven.Server/Documents/TransactionCommands/EnforceRevisionConfigurationCommand.cs

```diff
@@ -26,14 +26,24 @@ internal class EnforceRevisionConfigurationCommand : RevisionsScanningOperationC
     
         protected override long ExecuteCmd(DocumentsOperationContext context)
         {
-            MoreWork = false;
+            var idsToRemove = new List<string>();
             foreach (var id in _ids)
             {
                 _token.ThrowIfCancellationRequested();
-                _result.RemovedRevisions += (int)_revisionsStorage.EnforceConfigurationFor(context, id, _includeForceCreatedRevisionsOnDeleteInCaseOfNoConfiguration == false, ref MoreWork);
+                _result.RemovedRevisions += (int)_revisionsStorage.EnforceConfigurationFor(context, id, _includeForceCreatedRevisionsOnDeleteInCaseOfNoConfiguration == false, out var moreWork);
+
+                if (moreWork == false)
```

Comment:
this is for the case when we have many revisions for a given document and we remove it in batches? need to add a comment..
also if it is `true` why do keep iterating over the rest of the `_ids` ?

Reply:
For each document in the batch, we try to enforce (delete revisions according to its appropriate configuration).
If the doc has 'moreWork' (true), which means more revisions to delete (the revisions deletion has been stopped because the 'MaxUponDelete' property in the configuration), we keep its id on the 'ids' list - for trying enforce it again in a separate merged command. 
If the doc hasn't 'moreWork' (false), which means the revisions deletion has been finished, we remove its id from the 'ids' list.
We do that for all the docs in the 'ids' list.
After the command finished to be executed, we are taking the 'ids' list with the remaining ids (of the docs that their enforce wasn't finished because the had 'moreWork') and we are creating a new merged command with this list.
we are doing it until the 'ids' list is empty - this is happening in the 'RevisionsStorage.PerformRevisionsOperationOnSingleCollectionAsync' method (look at the internal while loop):
```
                    var moreWork = true;
                    while (moreWork)
                    {
                        token.Delay();
                        var cmd = createCommand(ids, result, token);
                        await _database.TxMerger.Enqueue(cmd);
                        moreWork = cmd.MoreWork;
                    }
```